### ISSUE TITLE
I got this error while training from a snapshot "Inputs to operation X of type Select must have the same size and shape"

### ISSUE THREAD

**Issue by kshehady** [fizyr/keras-retinanet#631]
Caused by op 'loss/classification_loss/Select_2', defined at:
  File "E:/M2/keras-retinanetmarwa/keras-retinanet/keras_retinanet/bin/train.py", line 350, in <module>
    main()
  File "E:/M2/keras-retinanetmarwa/keras-retinanet/keras_retinanet/bin/train.py", line 314, in main
    model            = keras.models.load_model(args.snapshot, custom_objects=custom_objects)
  File "D:\PythonEnv\AnacondaRetina\Anaconda3\lib\site-packages\keras\models.py", line 301, in load_model
    sample_weight_mode=sample_weight_mode)
  File "D:\PythonEnv\AnacondaRetina\Anaconda3\lib\site-packages\keras\engine\training.py", line 830, in compile
    sample_weight, mask)
  File "D:\PythonEnv\AnacondaRetina\Anaconda3\lib\site-packages\keras\engine\training.py", line 429, in weighted
    score_array = fn(y_true, y_pred)
  File "E:/M2/keras-retinanetmarwa/keras-retinanet/keras_retinanet/bin\..\..\keras_retinanet\losses.py", line 42, in _focal
    focal_weight = backend.where(keras.backend.equal(labels, 1), 1 - classification, classification)
  File "E:/M2/keras-retinanetmarwa/keras-retinanet/keras_retinanet/bin\..\..\keras_retinanet\backend\tensorflow_backend.py", line 46, in where
    return tensorflow.where(*args, **kwargs)
  File "D:\PythonEnv\AnacondaRetina\Anaconda3\lib\site-packages\tensorflow\python\ops\array_ops.py", line 2441, in where
    return gen_math_ops._select(condition=condition, t=x, e=y, name=name)
  File "D:\PythonEnv\AnacondaRetina\Anaconda3\lib\site-packages\tensorflow\python\ops\gen_math_ops.py", line 3987, in _select
    "Select", condition=condition, t=t, e=e, name=name)
  File "D:\PythonEnv\AnacondaRetina\Anaconda3\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "D:\PythonEnv\AnacondaRetina\Anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 2956, in create_op
    op_def=op_def)
  File "D:\PythonEnv\AnacondaRetina\Anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 1470, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Inputs to operation loss/classification_loss/Select_2 of type Select must have the same size and shape.  Input 0: [1,89685,55] != input 1: [1,89685,53]
	 [[Node: loss/classification_loss/Select_2 = Select[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"](loss/classification_loss/Equal, loss/classification_loss/sub_1, classification/concat)]]

**Comment by hgaiser** [comment]
What version of Keras are you using?

**Comment by hgaiser** [comment]
Closing due to inactivity.

**Comment by sebastianfrey** [comment]
I would like to reopen this issue. When I try to train from a previous snapshot, my model is loaded successfuly. After that step, `retinanet-train` crashes with a similar error message as described by the initiator of this issue:

![error-train-retinanet-from-snapshot](https://user-images.githubusercontent.com/15965420/54815767-f4f2cd80-4c92-11e9-859f-03363f91bc4e.png)

Hardware/Software:

GPU: Tesla P100 (16GB)
NVIDIA-Driver-Version: 418.43
CUDA-Version: 10.1
Docker-Version: 18.09.3
Tensorflow-Version: 1.12.0
Keras-Version: 2.2.4

Note, that I use Docker for training. This is my Dockerfile:

```Dockerfile
FROM tensorflow/tensorflow:1.12.0-gpu-py3
USER root

RUN apt-get update -y --fix-missing
RUN apt-get install -y \
	curl \
	git \
	libopencv-dev python-opencv \
        python-dev build-essential \
	nano vim

RUN pip3 install --upgrade pip
RUN pip3 install 'keras>=2.1.3'
RUN pip3 install --user --upgrade git+https://github.com/broadinstitute/keras-resnet
RUN pip3 install 'opencv-python==3.4.0.12'
RUN pip3 install python-dotenv pandas
RUN pip3 install pillow h5py
RUN pip3 install cython

RUN git clone https://github.com/fizyr/keras-retinanet /tmp/keras-retinanet
WORKDIR /tmp/keras-retinanet
RUN ls /tmp
RUN ln -s /usr/local/cuda/lib64/stubs/libcuda.so /usr/local/cuda/lib64/stubs/libcuda.so.1
RUN LD_LIBRARY_PATH=/usr/local/cuda/lib64/stubs/:$LD_LIBRARY_PATH python3 setup.py install
RUN rm /usr/local/cuda/lib64/stubs/libcuda.so.1
```

The command I use to initialize training from a snapshot:

```sh
retinanet-train \
  --batch-size 1 \
  --steps 231 \
  --image-max-side 500 \
  --random-transform \
  --tensorboard-dir /tmp/logs \
  --snapshot-path /tmp/snapshots  \
  --snapshot /tmp/backups/previous.h5 \
  csv annotations classes
```

If you need more information, please let me know.

**Comment by MoizTaimuri** [comment]
I also have the exact same issue

**Comment by MatthiasPilz** [comment]
I get a very similar issue (what sebastian highlighted in his image) when training on my own data and loading a snapshot that was pretrained on pascal. In some other forum somebody mentioned this has to do with the input size/shape but where do I have to adjust this?
